### ISSUE TITLE
player don't see number of FF points per turn

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#913]
it would be nice if the player could see the number of FF points he generates per turn. example in the screenshot.
![image](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/8c23706a-bd41-42d0-b556-d471accc7ec8)

**Comment by devolution79** [comment]
Hmm, perhaps a better place is the main interface just like the science bulbs of civ4 ?
Or perhaps not, since it could clutter the UI even more

**Comment by MrZorG33** [comment]
Yes, I also first thought about the interface on the global map. but then I realized that this information was not needed there all the time. but on the FF screen this information would be useful.

**Comment by raystuttgart** [comment]
There is no such thing as "FF **per turn**" ... 

"Military Points per turn" --> They are generated by combat Actions or by Buildings or by Events, not by turn.
"Exploratin Points per turn" --> They are geneared by Scouting, Events, ... not at the end of a turn.
"Religions Points per turn" --> They are also as it is also strongly generated by Missioning, Buildings ... but yes also by Crosses.
"Trade Points per turn" --> As they are generated by sales actions in Eurpe, by Buildings, by ... but not by turn.

How do you want to then create a "per turn" number if the whole feature does not work "per turn" ...
It is single player actions mostly that generate FF Points ...

Combat, Trading, Exploration, Missioning, Events, ...

The only exception is maybe "Political Points" ... but even there a lot of it is given by Buildings, Events ...

------------

**In other words:**
This will simply not work like you imagine as most of it is generated by Player actions ... that are **not "predictable"** ahead of time. Also there are so many places in the logic that can create FF Points ... that it would be incredible effort to calculate it.


There is no "single function" that calculates the FF Points.
The generation of FF points is distributed all over the player logic ... as it is driven by player actions mostly.
(And only some is generated regularly by e.g. Crosses or Liberty Bells.)

But these you will **never** be able to calculate at all per turn:
Military Points
Exploration Points
Trade Points

--------------

**Summary:**
It will technically **simply not work** ... as the feature does **not work based on turn** calculation.
(You cannot calculate something that is triggered by User Actions on a turn base ...)

The FF Points are e.g. added **instantly** when:

- Event Option is chosen
- Combat is resolved
- Yields are Traded
- Plots are Scouted
- Building is finished
- Improvement is built
- Mission is created
...

All of that happens in the logic of those features.
There is no chance to "predict" this for the next round ...

**Comment by MrZorG33** [comment]
I don't quite agree.

the number of religion points depends on the crosses. this means it can be calculated.
the number of policy points depends on the freedom bells. this can also be calculated.
the remaining categories of FF points can be converted from hammers, you didn't think about that. on small maps you have to get them this way. This means that this quantity can also be calculated.

otherwise I agree, the three categories largely depend on the player's actions. but these points can also be created in colonies.

**Comment by Schmiddie1979** [comment]
A little bit surprise in the game is simply part of the game. Not everything has to be calculated.

**Comment by raystuttgart** [comment]
@MrZorG33 
Reilgion Points only **partially** depends on the Crosses ... not fully.
Political Points only **partially** depends on Liberty Belles ... not fully.

And no, the rest can not be converted from Hammers ... as they have nothing to do with Hammers.
You simply cannot predict the Player actions ...

You can trust me ... as the programmer ... or not if I tell you it is **not** going to work correctly. :)
So whoever may want to give it a try ... good luck programming something like that. 👍 

The fact does not change: A lof of the stuff happens IN TURN by player actions that you can NOT fully predict it.
So you will never get the complete picture ... at most something that is a rough estimation ... 

**The best possible result:**
Some **vague number** that is **mostly incorrect** leading to **bug reports** telling that we calculate wrong.

But this will never be a feature that will truly give **reliable data** to help the player.
If you want to have an **"esitmation"** it is **already there**: As you can **see** turn by turn how the bar changed.

Still it is impossible to calculate an "exact number" ... as there are way too many features involved.
And also good luck to find somebody who will invest the effort. (Easily 30+ hours wasted ...)

**Comment by devolution79** [comment]
@raystuttgart is ofc correct in the sense that the FF points cannot be predicted with 100% accuracy. However, it could still be valuable to display the total number of political points produced in all cities somewhere

**Comment by raystuttgart** [comment]
I am careful honestly with "incomplete / incorrect information".

Because what will happen then is discussion and complaints:
- Why is ony Political Points shown and not the rest?
- I tested in my game and the actual number added was not the one shown.
- Why ... ?
- Why... ?
...

Trust me, it is not good to do stuff like that - as it will make users unhappy.
Rather show **nothing** than **incomplete** or **incorrect** information ...

But if you guys really want to start such discussions with our players and have pointless bug reports ... go ahead.
(But I am defnitely out of this discussion now ... it is kind of beating a horse that is already dead honestly.)

**Comment by victorxup** [comment]
A practical solution could be:
- Showing number of points **gained** since (or during) **last turn**.

It's no prediction, but still might give player a hint about what he wants to know. 
And probably much easier to implement.

**Comment by johan-buret** [comment]
> A practical solution could be:
> 
>     * Showing number of points **gained** since (or during) **last turn**.
> 
> 
> It's no prediction, but still might give player a hint about what he wants to know.

I have thought about it, victorxup, but even the most barebone implementation of this represent a significant coding effort.
